staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -242,6 +302,21 @@ func AddRequestsExecuting(ctx context.Context, priorityLevel, flowSchema string,
 	apiserverCurrentExecutingRequests.WithLabelValues(priorityLevel, flowSchema).Add(float64(delta))
 }
 
+// SetCurrentR sets the current-R (virtualTime) gauge for the given priority level
+func SetCurrentR(priorityLevel string, r float64) {
+	apiserverCurrentR.WithLabelValues(priorityLevel).Set(r)
+}
+
+// SetLatestS sets the latest-S (virtual time of dispatched request) gauge for the given priority level
+func SetDispatchMetrics(priorityLevel string, r, s, sMin, sMax, discountedSMin, discountedSMax float64) {
```

Comment:
This doesn't look safe if it is called concurrently from the same priorityLevel, is that possible? I suspect not. Please add a comment to the effect that that shouldn't/doesn't happen.

Reply:
I am not sure which sense of "safe" you are thinking about here.
The Prometheus library calls are safe for concurrent usage.
Perhaps you are thinking about consistency among the several metrics updated by `SetDispatchMetrics`.  This is not a new issue; we already have several places in the code that update a handful of metrics in a series of calls.  Unless and until the Prometheus library supports atomic updating of multiple metrics, we have no hope of such an atomic update.